src/result.rs

```diff
@@ -92,6 +94,7 @@ impl PartialEq for Error {
             (Value(e), Value(e2)) => e == e2,
             (Literal(e), Literal(e2)) => e == e2,
             (Interval(e), Interval(e2)) => e == e2,
+            (StringExt(e), StringExt(e2)) => e == e2,
```

Comment:
I.. felt this even from the first commit which in your pr, I appreciate your sense and skill to figure out structure and patterns in the code. It really looks that you are already familiar enough with the code base, awesome :)

Reply:
Thanks :), these well organized codes that help a lot to understand and make familiar with